pkg/scalers/azure_servicebus_scaler.go

```diff
@@ -233,105 +233,75 @@ func (s *azureServiceBusScaler) GetMetrics(ctx context.Context, metricName strin
 	return append([]external_metrics.ExternalMetricValue{}, metric), nil
 }
 
-type azureTokenProvider struct {
-	httpClient  *http.Client
-	ctx         context.Context
-	podIdentity kedav1alpha1.AuthPodIdentity
-}
-
-// GetToken implements TokenProvider interface for azureTokenProvider
-func (a azureTokenProvider) GetToken(uri string) (*auth.Token, error) {
-	ctx := a.ctx
-
-	var token azure.AADToken
-	var err error
-
-	switch a.podIdentity.Provider {
-	case kedav1alpha1.PodIdentityProviderAzure:
-		token, err = azure.GetAzureADPodIdentityToken(ctx, a.httpClient, a.podIdentity.IdentityID, serviceBusResource)
-	case kedav1alpha1.PodIdentityProviderAzureWorkload:
-		token, err = azure.GetAzureADWorkloadIdentityToken(ctx, a.podIdentity.IdentityID, serviceBusResource)
-	default:
-		err = fmt.Errorf("unknown pod identity provider")
-	}
-	if err != nil {
-		return nil, err
-	}
-
-	return auth.NewToken(auth.CBSTokenTypeJWT, token.AccessToken, token.ExpiresOn), nil
-}
-
 // Returns the length of the queue or subscription
 func (s *azureServiceBusScaler) getAzureServiceBusLength(ctx context.Context) (int64, error) {
-	// get namespace
-	namespace, err := s.getServiceBusNamespace(ctx)
+	// get adminClient
+	adminClient, err := s.getServiceBusAdminClient()
 	if err != nil {
 		return -1, err
 	}
 	// switch case for queue vs topic here
 	switch s.metadata.entityType {
 	case queue:
-		return getQueueEntityFromNamespace(ctx, namespace, s.metadata.queueName)
+		return getQueueLength(ctx, adminClient, s.metadata.queueName)
 	case subscription:
-		return getSubscriptionEntityFromNamespace(ctx, namespace, s.metadata.topicName, s.metadata.subscriptionName)
+		return getSubscriptionLength(ctx, adminClient, s.metadata.topicName, s.metadata.subscriptionName)
 	default:
 		return -1, fmt.Errorf("no entity type")
 	}
 }
 
 // Returns service bus namespace object
-func (s *azureServiceBusScaler) getServiceBusNamespace(ctx context.Context) (*servicebus.Namespace, error) {
-	var namespace *servicebus.Namespace
+func (s *azureServiceBusScaler) getServiceBusAdminClient() (*admin.Client, error) {
+	if s.client != nil {
+		return s.client, nil
+	}
+
+	var adminClient *admin.Client
 	var err error
 
 	switch s.podIdentity.Provider {
 	case "", kedav1alpha1.PodIdentityProviderNone:
-		namespace, err = servicebus.NewNamespace(servicebus.NamespaceWithConnectionString(s.metadata.connection))
+		adminClient, err = admin.NewClientFromConnectionString(s.metadata.connection, nil)
 		if err != nil {
-			return namespace, err
+			return nil, err
 		}
 	case kedav1alpha1.PodIdentityProviderAzure, kedav1alpha1.PodIdentityProviderAzureWorkload:
-		namespace, err = servicebus.NewNamespace()
+		credentials, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
I don't think this works. I found this issue https://github.com/Azure/azure-sdk-for-go/issues/15615.

Reply:
In .NET the ManagedIdentity credentials also work for Kubernetes Workload Federation (if I recall correctly from the KEDA .NET samples I added). However, it doesn't seem to be added for Golang yet.

Reply:
hum... I'll try to apply their suggestion as an extra token source, at least till they officially support workload identity in v1.2.0 Basically because I'd like to unlock the SDK migration

Reply:
Yup, in fact I think using official SDKs wherever possible instead of manually making REST API calls would be beneficial.

Reply:
Do check the workload identity files, I already had to implement a number of token provider interfaces, so you "might" not have to redefine them.

Reply:
PTAL, I have done a temporal implementation of default client provider till they include wi in the sdk.
Once it's supported, I'd update it and use SDK for more scalers instead of HTTP request